runtime/ms-rest-azure/lib/filters/rpRegistrationFilter.js

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information. 
+
+'use strict';
+
+var request = require('request');
+var utils = require('../utils');
+
+exports.create = function create() {
+  return function handle (options, next, callback) {
+    return next(options, function (err, response, body) {
+      if (err) {
+        return callback(err);
+      }
+      let rpName, urlPrefix;
+      if (response.statusCode === 409) {
+        rpName = exports.checkRPNotRegisteredError(body);
+        if (rpName) {
+          urlPrefix = exports.extractSubscriptionUrl(options.url);
+          return exports.registerRP(urlPrefix, rpName, options, (err, result) => {
+            if (err) {
+              return callback(err);
+            }
+            if (result) {
+              //retry the original request
+              options.headers['x-ms-client-request-id'] = utils.generateUuid();
+              return request(options.url, options, (error, response, body) => {
+                return callback(error, response, response.body);
+              });
+            } else {
+              return callback(err, response, body);
+            }
+          });
+        }
+      }
+      return callback(err, response, body);
+    });
+  };
+};
+
+exports.checkRPNotRegisteredError = function checkRPNotRegisteredError(body) {
+  let result, responseBody;
+  if (body) {
+    try {
+      responseBody = JSON.parse(body);
+    } catch (err) {
+      //do nothing;

```

Comment:
might want to return `false` here. otherwise you'd be potentially dotting off of a `null responseBody` after this line. i.e, i'm wondering if the first check after this `responseBody.error` can possibly throw.